src/common/transformations/src/transformations/common_optimizations/optimize_strided_slice.cpp

```diff
@@ -366,25 +368,72 @@ bool ov::pass::GroupedSliceToVSplitOptimization::run_on_model(const std::shared_
     return graph_rewritten;
 }
 
+ov::pass::SliceSequenceToSingleSlice::SliceSequenceToSingleSlice() {
+    MATCHER_SCOPE(SliceSequenceToSingleSlice);
+    using namespace ov::op;
+    using namespace ov::op::util;
+    using namespace ov::pass::pattern;
+
+    auto const_axes_1_pattern = wrap_type<v0::Constant>();
+    auto const_axes_2_pattern = wrap_type<v0::Constant>();
+    auto slice_1_pattern =
+        wrap_type<v8::Slice>({any_input(), any_input(), any_input(), any_input(), const_axes_1_pattern},
+                             consumers_count(1));
+    auto slice_2_pattern =
+        wrap_type<v8::Slice>({slice_1_pattern, any_input(), any_input(), any_input(), const_axes_2_pattern});
+
+    ov::matcher_pass_callback callback = [=](Matcher& m) {
+        const auto& pattern_to_output = m.get_pattern_map();
+        auto slice_1 = pattern_to_output.at(slice_1_pattern);
+        auto slice_2 = pattern_to_output.at(slice_2_pattern);
+
+        auto const_axes_1 = ov::as_type_ptr<v0::Constant>(pattern_to_output.at(const_axes_1_pattern));
+        auto const_axes_2 = ov::as_type_ptr<v0::Constant>(pattern_to_output.at(const_axes_2_pattern));
+
+        auto axes_1_values = const_axes_1->cast_vector<int64_t>();
+        auto axes_2_values = const_axes_2->cast_vector<int64_t>();
+
+        // supported a simple scenario when the axes_1 values and axes_2 values don't intersect.
+        for (const auto& axis : axes_1_values) {
+            if (std::find(axes_2_values.begin(), axes_2_values.end(), axis) != axes_2_values.end()) {
+                return false;
+            }
+        }
+
+        auto begin = std::make_shared<v0::Concat>(OutputVector{slice_1->input_value(1), slice_2->input_value(1)}, 0);
+        auto end = std::make_shared<v0::Concat>(OutputVector{slice_1->input_value(2), slice_2->input_value(2)}, 0);
+        auto step = std::make_shared<v0::Concat>(OutputVector{slice_1->input_value(3), slice_2->input_value(3)}, 0);
+        auto axes = std::make_shared<v0::Concat>(OutputVector{slice_1->input_value(4), slice_2->input_value(4)}, 0);
+        auto one_slice = std::make_shared<ov::op::v8::Slice>(slice_1->input_value(0),
+                                                             try_fold_unary_output(begin),
```

Comment:
is folding needed here? we have ConstantFolding transformation in MOC

Reply:
it's required